glusterd2/commands/volumes/brick-replace.go

```diff
@@ -0,0 +1,221 @@
+package volumecommands
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/bricksplanner"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	gderrors "github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func registerReplaceBrickStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"brick-replace.PrepareBricks", prepareBricks},
+		{"brick-replace.ReplaceVolinfo", replaceVolinfo},
+		{"brick-replace.StartBrick", startBrick},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func replaceBrickHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	volname := mux.Vars(r)["volname"]
+
+	if !volume.IsValidName(volname) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrInvalidVolName)
+		return
+	}
+
+	// Get Volume Info
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	// Unmarshal ReplaceBrickReq
+	var req api.ReplaceBrickReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrJSONParsingFailed)
+		return
+	}
+
+	if uuid.Parse(req.SrcPeerID) == nil {
```

Comment:
this also needs to be done before  accessing data  from ETCD

Reply:
change Invalid to invalid

Reply:
What kind of validation are you looking for?

Reply:
input request validation `req.SrcPeerID` is valid UUID or not, need to be done before checking the volume present in the cluster or not. if the user sends parallel invalid requests we should not exhaust the ETCD resources, the input validation in the early stage avoid this kind of issues.